### ISSUE TITLE
Annotated attributes in Enum should be treated as non-members even in stubs

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#8061]
The typing spec indicates that when an enum definition is found in a type stub, any attributes that are annotated with an explicit type annotation should be treated as non-members. Pyright doesn't currently do this — mainly because typeshed stubs violated this rule until recently.

```python
class Pet(Enum):
    genus: str  # Non-member attribute
    species: str  # Non-member attribute

    CAT = ...  # Member attribute
    DOG = ...  # Member attribute
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.366.
